auraed/src/runtime/cells/cell.rs

```diff
@@ -41,25 +72,28 @@ impl Cell {
         Self { name, cgroup, executables: Default::default() }
     }
 
-    pub fn free(self) -> Result<()> {
-        self.cgroup.delete()?;
+    pub fn free(self) -> Result<(), CellError> {
+        self.cgroup.delete().map_err(|e| CellError::FailedToFree {
+            cell_name: self.name.clone(),
+            source: e,
+        })?;
+
         Ok(())
     }
 
-    pub fn spawn_executable(
+    pub fn start_executable(
         &mut self,
         exe_name: ExecutableName,
         mut command: Command,
         args: Vec<String>,
         _description: String,
-    ) -> Result<()> {
+    ) -> Result<(), CellError> {
         // Check if there was already an executable with the same name.
         if self.executables.contains_key(&exe_name) {
-            return Err(anyhow!(
-                "executable '{exe_name}' already exists in {}",
-                self.name
-            )
-            .into());
+            return Err(CellError::ExecutableExists {
+                cell_name: self.name.clone(),
+                executable_name: exe_name,
+            });
         }
 
         let _ = command.args(args);
```

Comment:
this assumes that command.args doesn't return a new instance of a Command.  is that assuming too much about the implementation?

Reply:
Yup, that's a great point. This can break tomorrow and we wouldn't know